sphinx/transforms/__init__.py

```diff
@@ -327,3 +329,33 @@ def visit_pending_xref(self, node):
     def visit_image(self, node):
         # type: (nodes.Node) -> None
         raise nodes.SkipNode
+
+
+class SphinxSmartQuotes(DocutilsSmartQuotes):  # NOQA
+    """
+    Customized SmartQuotes to avoid transform for some extra node types.
+    """
+    def get_tokens(self, txtnodes):
+        # A generator that yields ``(texttype, nodetext)`` tuples for a list
+        # of "Text" nodes (interface to ``smartquotes.educate_tokens()``).
+
+        texttype = {True: 'literal',  # "literal" text is not changed:
+                    False: 'plain'}
+        for txtnode in txtnodes:
+            nodetype = texttype[isinstance(txtnode.parent,
+                                           (nodes.literal,
+                                            nodes.literal_block,
+                                            addnodes.literal_emphasis,
+                                            addnodes.literal_strong,
+                                            addnodes.desc_signature,
+                                            addnodes.productionlist,
+                                            addnodes.desc_optional,
+                                            addnodes.desc_name,
```

Comment:
@tk0miya I realize the situation is not fully satisfying. The 1.5.6 had a ``no_smarty`` flag, hence an inheritance process which prevented smart quotes in ``desc_addname`` nodes for example, as they were children of ``desc_signature``. Unfortunately the Docutils SmartQuotes tests only the parent node. Thus yesterday I forgot to add ``desc_addname``. Some refactoring is thus needed for our SphinxSmartQuotes to avoid smart quotes in all sub-nodes of ``desc_signature``. At least I should add ``desc_addname``, but this ever-increasing test of the sole parent does not look satisfying.

Nevertheless I will push that on stable. Then better refactoring can take place.